controllers/atlasmigration_controller_test.go

```diff
@@ -0,0 +1,406 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ariga/atlas-operator/api/v1alpha1"
+	"github.com/ariga/atlas-operator/internal/atlas"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// migrateCLIMock is a mock implementation of the MigrateCLI interface
+type migrateCLIMock struct {
+	mock.Mock
+}
+
+func (m *migrateCLIMock) Apply(ctx context.Context, data *atlas.ApplyParams) (*atlas.ApplyReport, error) {
+	args := m.Called(ctx, data)
+	return args.Get(0).(*atlas.ApplyReport), args.Error(1)
+}
+
+func (m *migrateCLIMock) Status(ctx context.Context, data *atlas.StatusParams) (*atlas.StatusReport, error) {
+	args := m.Called(ctx, data)
+	return args.Get(0).(*atlas.StatusReport), args.Error(1)
+}
+
+// newAtlasMigrationReconcilerMock returns a AtlasMigrationReconciler with a fake client
+// please create a new namespace for your suite tests
+func mockMigrationReconciler(ns string, objs ...client.Object) *AtlasMigrationReconciler {
+	scheme := runtime.NewScheme()
+	v1alpha1.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	switch ns {
+	case "default":
```

Comment:
why only check `default` namespace?
```suggestion
	switch ns {
	case "default":
```

Reply:
removed, but the idea behide is producing local env by args